src/App.jsx

```diff
@@ -14,70 +15,67 @@ export const goods = [
   'Garlic',
 ];
 
-export const App = () => (
-  <main className="section container">
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
+export const App = () => {
+  const [selectedGood, setSelectedGood] = useState('Jam');
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
+  const clearSelection = () => {
+    setSelectedGood('');
+  };
 
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {selectedGood ? `${selectedGood} is selected` : 'No goods selected'}
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={clearSelection}
+          />
+        )}
+      </h1>
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              key={good}
+              data-cy="Good"
+              className={selectedGood === good
+                ? 'has-background-success-light'
+                : ''}
             >
-              +
-            </button>
-          </td>
+              <td>
+                {selectedGood !== good && (
+                  <button
+                    data-cy="AddButton"
+                    type="button"
+                    className="button"
+                    onClick={() => setSelectedGood(good)}
+                  >
+                    +
+                  </button>
+                )}
+                {selectedGood === good && (
+                  <button
+                    data-cy="RemoveButton"
+                    type="button"
+                    className="button is-info"
+                    onClick={clearSelection}
+                  >
+                    -
+                  </button>
```

Comment:
I think that at this moment you repeat yourself, you can change data-cy, className, plus and minus signs by using ternary operators 
For example: 
const isSelected = good === selectedGood;
```suggestion
              <td>
                  <button
                      data-cy={isSelected
                        ? 'RemoveButton'
                        : 'AddButton'
                      }
                      type="button"
                      className={`button ${isSelected
                        ?  ' is-info'
                        :  ''
                      '}`}
                  >
                    {isSelected ? '-' : '+'}
                  </button>
                )}
      
```

Reply:
in general you are absolutely right
but in this case the code is more difficult to read and takes more time to understand
since the components are small, better to leave the decision of Maria